pkg/translator/prometheusremotewrite/helper_v2.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package prometheusremotewrite // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheusremotewrite"
+
+import (
+	"github.com/prometheus/common/model"
+	writev2 "github.com/prometheus/prometheus/prompb/io/prometheus/write/v2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	conventions "go.opentelemetry.io/otel/semconv/v1.25.0"
+
+	prometheustranslator "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheus"
+)
+
+// addResourceTargetInfo converts the resource to the target info metric.
+func addResourceTargetInfoV2(resource pcommon.Resource, settings Settings, timestamp pcommon.Timestamp, converter *prometheusConverterV2) {
+	if settings.DisableTargetInfo || timestamp == 0 {
+		return
+	}
+
+	attributes := resource.Attributes()
+	identifyingAttrs := []string{
+		string(conventions.ServiceNamespaceKey),
+		string(conventions.ServiceNameKey),
+		string(conventions.ServiceInstanceIDKey),
+	}
+	nonIdentifyingAttrsCount := attributes.Len()
+	for _, a := range identifyingAttrs {
+		_, haveAttr := attributes.Get(a)
+		if haveAttr {
+			nonIdentifyingAttrsCount--
+		}
+	}
+	if nonIdentifyingAttrsCount == 0 {
+		// If we only have job + instance, then target_info isn't useful, so don't add it.
+		return
+	}
+
+	name := prometheustranslator.TargetInfoMetricName
+	if len(settings.Namespace) > 0 {
+		// TODO what to do with this in case of full utf-8 support?
+		name = settings.Namespace + "_" + name
```

Comment:
@ywwg fyi

Reply:
basically with full utf-8 support enabled if someone uses the namespace feature do add `my.ns` dp we want to join and create `my.ns_target_info`?

Reply:
Ah ok, got it!


There's another problem, actually, that makes this UTF-8 problem nonexistent! I believe we shouldn't add namespace to the `target_info` metric at all, because Prometheus is building integration based on that specific metric name, e.g. the `info()` function. If a namespace is added, it breaks the assumption that Prometheus has on OTel data

Reply:
People often use namespace to prevent metrics from colliding with each other.  IMO people shouldn't ever use namespace, and should use external labels instead.  But if we are going to offer namespace configuration, it should apply to all metrics

Reply:
My 2 cents:
Even the spec assumes target_info is hardcoded  (https://opentelemetry.io/docs/specs/otel/compatibility/prometheus_and_openmetrics/#resource-attributes-1) so I would consider dropping support adding namespace to the metric in both RW1 and RW2 in a separate PR, I don't think we should blog this PR on it, or at least we can resolve this with opening an issue for it / agreeing to discuss this in the SIG call / slack etc.

What do you think @dashpole @ArthurSens

Reply:
okay resolving this.